File: designpattern/src/main/java/com/seniorlibs/designpattern/ch25v2/repository/MetricsStorage.java
package com.seniorlibs.designpattern.ch25v2.repository;

import com.seniorlibs.designpattern.ch25v2.model.RequestInfo;
import java.util.List;
import java.util.Map;

public interface MetricsStorage {

    void saveRequestInfo(RequestInfo requestInfo);

    List<RequestInfo> getRequestInfos(String apiName, long startTimeInMillis, long endTimeInMillis);

    Map<String, List<RequestInfo>> getRequestInfos(long startTimeInMillis, long endTimeInMillis);
}

File: designpattern/src/main/java/com/seniorlibs/designpattern/ch25v3/repository/MetricsStorage.kt
package com.seniorlibs.designpattern.ch25v3.repository

import com.seniorlibs.designpattern.ch25v3.model.RequestInfo

interface MetricsStorage {
    fun saveRequestInfo(requestInfo: RequestInfo)

    fun getRequestInfos(
        apiName: String,
        startTimeInMillis: Long,
        endTimeInMillis: Long
    ): List<RequestInfo>

    fun getRequestInfos(
        startTimeInMillis: Long,
        endTimeInMillis: Long
    ): Map<String, List<RequestInfo>>
}